backend/de.metas.business.rest-api-impl/src/main/java/de/metas/rest_api/bpartner/relation/BPRelationsService.java

```diff
@@ -0,0 +1,229 @@
+{
+
+	private final JsonServiceFactory jsonServiceFactory;
+	private final IBPRelationDAO bpRelationDAO = Services.get(IBPRelationDAO.class);
+
+	public BPRelationsService(@NonNull final JsonServiceFactory jsonServiceFactory)
+	{
+		this.jsonServiceFactory = jsonServiceFactory;
+	}
+
+	public JsonResponseBPRelationComposite getRelationsForPartner(@NonNull final OrgId orgId, @NonNull final IdentifierString bpartnerIdentifier)
+	{
+		final Optional<BPartnerComposite> bPartnerComposite = jsonServiceFactory.createRetriever().getBPartnerComposite(orgId, bpartnerIdentifier);
+
+		final Collection<JsonResponseBPRelationItem> relations = new ArrayList<>();
+		bPartnerComposite.ifPresent(rel -> relations.addAll(getRelationsForBPartner(orgId, rel.getBpartner().getId())));
+		return JsonResponseBPRelationComposite.builder()
+				.responseItems(relations)
+				.build();
+	}
+
+	private Collection<JsonResponseBPRelationItem> getRelationsForBPartner(final @NonNull OrgId orgId, @NonNull final BPartnerId id)
+	{
+		final Stream<BPRelation> relationsForBpartner = bpRelationDAO.getRelationsForBpartner(orgId, id);
+		return relationsForBpartner.map(this::toJson).collect(Collectors.toList());
+	}
+
+	private JsonResponseBPRelationItem toJson(@NonNull final BPRelation relation)
+	{
+		final JsonResponseBPRelationItem.JsonResponseBPRelationItemBuilder builder = JsonResponseBPRelationItem.builder()
+				.bpartnerId(relation.getBpartnerId().toJson())
+				.targetBPartnerId(relation.getTargetBPartnerId().toJson())
+				.targetBPLocationId(relation.getTargetBPLocationId().getRepoId())
+				.name(relation.getName())
+				.description(relation.getDescription())
+				.billTo(relation.getBillTo())
+				.fetchedFrom(relation.getFetchedFrom())
+				.handoverLocation(relation.getHandoverLocation())
+				.payFrom(relation.getPayFrom())
+				.remitTo(relation.getRemitTo())
+				.shipTo(relation.getShipTo())
+				.active(relation.getActive());
+		final BPRelation.BPRelationRole role = relation.getRole();
+		if (role != null)
+		{
+			builder.role(JsonBPRelationRole.valueOf(role.name()));
+		}
+		final BPartnerLocationId bpLocationId = relation.getBpLocationId();
+		if (bpLocationId != null)
+		{
+			builder.locationId(bpLocationId.getRepoId());
+		}
+		final ExternalId externalId = relation.getExternalId();
+		if (externalId != null)
+		{
+			builder.externalId(JsonExternalId.ofOrNull(externalId.getValue()));
+		}
+
+		return builder.build();
+	}
+
+	public void createOrUpdateRelations(@NonNull final OrgId orgId,
+			@NonNull final IdentifierString bpartnerIdentifier,
+			@Nullable final IdentifierString locationIdentifier,
+			@NonNull final List<JsonRequestBPRelationTarget> relatesTo)
+	{
+		final Optional<BPartnerComposite> bPartnerComposite = jsonServiceFactory.createRetriever().getBPartnerComposite(orgId, bpartnerIdentifier);
+		final BPartnerComposite bpartner = Check.assumePresent(bPartnerComposite, "No bpartner found for identifier: " + bpartnerIdentifier);
```

Comment:
please throw a MissingResourceException

Reply:
Changed to throw MissingResourceException